bench/tiered/push_pull.c

```diff
@@ -0,0 +1,271 @@
+    if (opts->tiered_storage) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/bucket", home));
+        testutil_make_work_dir(buf);
+    }
+
+    testutil_wiredtiger_open(opts, home, conn_config, NULL, &conn, false);
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create and populate table. Checkpoint the data after that. */
+    testutil_check(session->create(session, opts->uri, table_config_row));
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), flush ? "flush_tier=(enabled,force=true)" : ""));
+
+    gettimeofday(&start, 0);
+    populate(session, num_records);
+
+    testutil_check(session->checkpoint(session, buf));
+    gettimeofday(&end, 0);
+
+    diff_sec = difftime_sec(start, end);
+
+    sleep(1);
+
+    get_file_size(home, &file_size);
+    testutil_assert(diff_sec > 0);
+    printf("Bytes Written - %" PRIi64 ", Time took - %f seconds, Throughput - %f MB/second\n",
+      file_size, diff_sec, ((file_size / diff_sec) / WT_MEGABYTE));
+}
+
+/*
+ * populate --
+ *     Populate the table.
+ */
+static void
+populate(WT_SESSION *session, uint64_t num_records)
+{
+    WT_CURSOR *cursor;
+    WT_RAND_STATE rnd;
+    uint64_t i, str_len;
+
+    __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
+
+    str_len = sizeof(data_str) / sizeof(data_str[0]);
+    for (i = 0; i < str_len - 1; i++)
+        data_str[i] = 'a' + (uint32_t)__wt_random(&rnd) % 26;
+
+    data_str[str_len - 1] = '\0';
+
+    testutil_check(session->open_cursor(session, opts->uri, NULL, NULL, &cursor));
+    for (i = 0; i < num_records; i++) {
+        cursor->set_key(cursor, i + 1);
+        cursor->set_value(cursor, data_str);
+        testutil_check(cursor->insert(cursor));
+    }
+
+    testutil_check(cursor->close(cursor));
+    cursor = NULL;
+}
+
+/*
+ * compute_tiered_file_size --
+ *     Iterate over all the tiered files and compute file size..
+ */
+static void
+compute_tiered_file_size(const char *home, const char *tablename, int64_t *file_size)
+{
+    char stat_path[512];
+    int index;
+    struct stat stats;
+
+    for (index = 2; index < MAX_TIERED_FILES; ++index) {
+        testutil_check(__wt_snprintf(
+          stat_path, sizeof(stat_path), "%s/%s-%10.10d.wtobj", home, tablename, index));
+
+        /* Return if the stat fails that means the file does not exist. */
+        if (stat(stat_path, &stats) == 0)
+            *file_size += stats.st_size;
+        else
+            return;
+    }
+}
+
+/*
+ * get_file_size --
+ *     Retrieve the file size of the table.
+ */
+static void
+get_file_size(const char *home, int64_t *file_size)
+{
+    struct stat stats;
+
+    char stat_path[512];
+    const char *tablename;
+
+    tablename = strchr(opts->uri, ':');
+    testutil_assert(tablename != NULL);
+    tablename++;
+
+    if (opts->tiered_storage)
+        testutil_check(
+          __wt_snprintf(stat_path, sizeof(stat_path), "%s/%s-0000000001.wtobj", home, tablename));
```

Comment:
See previous comment: all tiered storage files from 00001 on up should be looked at in `compute_tiered_file_size()`.

Reply:
Fixed